omod/src/test/java/org/openmrs/module/registrationapp/page/controller/RegisterPatientPageControllerTest.java

```diff
@@ -0,0 +1,163 @@
+        Mockito.when(Context.getRegisteredComponent(AddressSupportCompatibility.ID, AddressSupportCompatibility.class)).thenReturn(addressSupport);
+
+        UserContext userContext = Mockito.mock(UserContext.class);
+        Context.setUserContext(userContext);
+        Mockito.when(userContext.isAuthenticated()).thenReturn(true);
+
+        pageModel = new PageModel();
+
+        ObjectNode objectNode = new ObjectNode(null);
+        objectNode.put("sections", new ArrayNode(null));
+        Mockito.when(jsonNode.isNull()).thenReturn(false);
+        Mockito.when(jsonNode.getBooleanValue()).thenReturn(true);
+        objectNode.put("registrationEncounter", jsonNode);
+        objectNode.put("allowRetrospectiveEntry", jsonNode);
+        objectNode.put("allowUnknownPatients", jsonNode);
+        objectNode.put("allowManualIdentifier", jsonNode);
+        objectNode.put("patientDashboardLink", jsonNode);
+        Mockito.when(app.getConfig()).thenReturn(objectNode);
+
+        Mockito.when(emrApiProperties.getPrimaryIdentifierType()).thenReturn(patientIdentifierType);
+
+        patientIdentifierType = new PatientIdentifierType();
+        patientIdentifierType.setId(1);
+
+        breadcrumbOverride = "home>sample";
+
+        List<Extension> extensions = new ArrayList<Extension>();
+        extensions.add(new Extension());
+        Mockito.when(appFrameworkService.getExtensionsForCurrentUser("registerPatient.includeFragments")).thenReturn(extensions);
+
+        registerPatientPageController = new RegisterPatientPageController();
+    }
+
+    @Test
+    public void get_shouldCreateANewPatientWhenPatientIsNull() throws Exception {
+        // Fixture setup
+        Patient existingPatient = null;
+
+        // Execution
+        registerPatientPageController.get(
+                uiSessionContext,
+                pageModel,
+                app,
+                existingPatient,
+                breadcrumbOverride,
+                existingPatient,
+                emrApiProperties,
+                appFrameworkService,
+                ui);
+
+        // Assertion
+        Patient result = (Patient) pageModel.getAttribute("patient");
+        Assert.assertNotNull(result);
+        Assert.assertNull(result.getId());
+    }
+
+    @Test
+    public void get_shouldUseExistingPatientWhenPatientIsNotNull() throws Exception {
+        // Fixture setup
+        Patient existingPatient = new Patient();
+        existingPatient.setId(10);
+
+        // Execution
+        registerPatientPageController.get(
+                uiSessionContext,
+                pageModel,
+                app,
+                existingPatient,
+                breadcrumbOverride,
+                existingPatient,
+                emrApiProperties,
+                appFrameworkService,
+                ui);
+
+        // Assertion
+        Patient result = (Patient) pageModel.getAttribute("patient");
+        Assert.assertEquals(existingPatient.getId(), result.getId());
```

Comment:
I understand that you are trying to get 2 checks done in assert.equal check. It would be a better coding convention and improve readability if there was assert.null right before the assert.equals function.